winpkg/ddev_windows_installer.nsi

```diff
@@ -0,0 +1,1657 @@
+                StrCpy $R0 "$R0|"
+            ${EndIf}
+            StrCpy $R0 "$R0$R3"
+        ${EndIf}
+
+        next_key:
+        IntOp $R1 $R1 + 1
+    ${EndWhile}
+
+    DetailPrint "Registry enumeration complete. Final list: [$R0]"
+    Push $R0
+FunctionEnd
+
+; TODO: there seem to be missing error checks here.
+Function InstallWSL2CommonSetup
+    ; Check for WSL2
+    DetailPrint "Checking WSL2 version..."
+    nsExec::ExecToStack 'wsl.exe -l -v'
+    Pop $1
+    Pop $0
+    DetailPrint "WSL version check output: $0"
+    DetailPrint "WSL version check exit code: $1"
+    ${If} $1 != 0
+        DetailPrint "ERROR: WSL2 not detected - exit code: $1, output: $0"
+        Push "WSL2 does not seem to be installed. Please install WSL2 and Ubuntu before running this installer."
+        Call ShowErrorAndAbort
+    ${EndIf}
+
+    ; Check for Ubuntu in selected distro
+    DetailPrint "Checking selected distro $SELECTED_DISTRO..."
+    nsExec::ExecToStack 'wsl -d $SELECTED_DISTRO bash -c "cat /etc/os-release | grep -i ^NAME="'
+    Pop $1
+    Pop $0
+    ${If} $1 != 0
+        DetailPrint "ERROR: Cannot access distro $SELECTED_DISTRO - exit code: $1, output: $0"
+        Push "Could not access the selected distro. Please ensure it's working properly."
+        Call ShowErrorAndAbort
+    ${EndIf}
+
+    ; Check for WSL2 kernel
+    DetailPrint "Checking WSL2 kernel..."
+    nsExec::ExecToStack 'wsl -d $SELECTED_DISTRO uname -v'
+    Pop $1
+    Pop $0
+    DetailPrint "WSL kernel version: $0"
+    ${If} $1 != 0
+        DetailPrint "ERROR: WSL version check failed - exit code: $1, output: $0"
+        Push "Could not check WSL version. Please ensure WSL is working."
+        Call ShowErrorAndAbort
+    ${EndIf}
+    ${If} $0 == ""
+        DetailPrint "ERROR: Empty WSL version output"
+        Push "Could not detect WSL version. Please ensure WSL is working."
+        Call ShowErrorAndAbort
+    ${EndIf}
+    ${If} $0 == "WSL"
+        DetailPrint "ERROR: WSL1 detected instead of WSL2 - version output: $0"
+        Push "The selected distro ($SELECTED_DISTRO) is not WSL2. Please use a WSL2 distro."
+        Call ShowErrorAndAbort
+    ${EndIf}
+    DetailPrint "WSL2 detected successfully."
+
+    ; Check for non-root default user
+    DetailPrint "Checking for non-root user..."
+    nsExec::ExecToStack 'wsl -d $SELECTED_DISTRO whoami'
+    Pop $1  ; error code
+    Pop $0  ; output
+    DetailPrint "Current user: $0"
+    ${If} $1 != 0
+        DetailPrint "ERROR: WSL user check failed - exit code: $1, output: $0"
+        Push "Could not check WSL user. Please ensure WSL is working."
+        Call ShowErrorAndAbort
+    ${EndIf}
+    ${If} $0 == "root"
+        DetailPrint "ERROR: Default WSL user is root - this is not supported"
+        Push "The default user in your WSL2 distro is root. Please configure an ordinary default user."
+        Call ShowErrorAndAbort
+    ${EndIf}
+    DetailPrint "Non-root user detected successfully."
```

Comment:
I think there is a problem with trailing lines in the output, the check for `root` doesn't catch it:

![image](https://github.com/user-attachments/assets/711ca72e-02af-4d4f-9acf-2396b23199a1)

Reply:
To reproduce, create a new WSL2, but do not boot it, i.e. do not create a new user.